### PR TITLE
Add workflow to automatically update iOS and Android dependencies.

### PR DESCRIPTION
**Pull request by jonsimantov**
This workflow will:
- Check out the repo and create a branch
- Update the iOS and/or Android dependencies in that branch
- Push the branch to GitHub
- Create a PR on the branch
- Trigger integration tests to run on the PR (via label)

For now, to use, go to "Actions", select the "Update Android and iOS dependencies" workflow, and use workflow_dispatch to run the workflow. In most cases, the default inputs should be fine (after this PR goes in).

### CODE REVIEW

**Comment by jonsimantov** [comment]
Sample PR created by this workflow: https://github.com/firebase/firebase-cpp-sdk/pull/635

**Comment by jonsimantov** [comment]
This is now ready for review!

Sample run: https://github.com/firebase/firebase-cpp-sdk/actions/runs/1215602717
Sample PR created by this workflow: https://github.com/firebase/firebase-cpp-sdk/pull/645

**Review by DellaBitta — CHANGES_REQUESTED** [review]
Double check admob's dependency version.

File: Android/firebase_dependencies.gradle [hunk 1]
```diff
@@ -17,7 +17,7 @@ import org.gradle.util.ConfigureUtil;
 // A map of library to the dependencies that need to be added for it.
 def firebaseDependenciesMap = [
   'app' : ['com.google.firebase:firebase-analytics:19.0.1'],
-  'admob' : ['com.google.firebase:firebase-ads:19.8.0',
+  'admob' : ['com.google.firebase:firebase-ads:19.4.0',
```
[the remark this answers is not in the document]
Comment (jonsimantov) [reply Android/firebase_dependencies.gradle:20]: No, was being used for testing - reverting.

File: .github/workflows/update-dependencies.yml [hunk 2]
```diff
@@ -1,13 +1,133 @@
...
+          echo "UPDATE_LOGFILE=update_log.txt" >> $GITHUB_ENV
+
+      - name: Run update script
+        run: |
+          if [[ ${{ github.event.inputs.updateiOS }} -eq 1 ]]; then
+            if [[ ${{ github.event.inputs.updateAndroid }} -eq 1 ]]; then
+              # Update both
+              echo "Updating all dependencies"
+              python scripts/update_android_ios_dependencies.py --logfile=${UPDATE_LOGFILE}
+              echo "CHOSEN_DEPS=" >> $GITHUB_ENV
```
[the remark this answers is not in the document]
Comment (jonsimantov) [reply .github/workflows/update-dependencies.yml:67]: How about "Android and iOS"? :)

File: .github/workflows/update-dependencies.yml [hunk 3]
```diff
@@ -1,13 +1,133 @@
...
+      - name: Install prerequisites
+        run: |
+          python scripts/gha/install_prereqs_desktop.py
+          python -m pip install requests
+          
+      - name: Name new branch
+        run: |
+          date_str=$(date "+%Y%m%d-%H%M%S")
+          echo "NEW_BRANCH=${{env.branchPrefix}}${{github.run_number}}-${date_str}" >> $GITHUB_ENV
+      - name: Create new branch
```
[the remark this answers is not in the document]
Comment (jonsimantov) [reply .github/workflows/update-dependencies.yml:55]: Fixed.

File: .github/workflows/update-dependencies.yml [hunk 4]
```diff
@@ -1,13 +1,133 @@
...
+            fi
+            # iOS: Update Firestore external version to match Firestore Cocoapod version.
+            firestore_version=$(grep "pod 'Firebase/Firestore'" ios_pod/Podfile | sed "s/.*'\([0-9][0-9]*\.[0-9][0-9]*\.[0-9][0-9]*\)'.*/\1/")
+            sed -i~ "s/^set(version [^)]*)/set(version CocoaPods-${firestore_version})/i" cmake/external/firestore.cmake
+          elif [[ ${{ github.event.inputs.updateAndroid }} -eq 1 ]]; then
+            # Update Android only
+              echo "Updating Android dependencies only"
+              python scripts/update_android_ios_dependencies.py --skip_ios --logfile=${UPDATE_LOGFILE}
+              echo "CHOSEN_DEPS=Android " >> $GITHUB_ENV
+          fi
```
[the remark this answers is not in the document]
Comment (jonsimantov) [reply .github/workflows/update-dependencies.yml:86]: Sure, why not. Added.

File: scripts/gha/create_pull_request.py [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+import datetime
+import shutil
+
+from absl import app
+from absl import flags
+from absl import logging
+
+import github
+
+FLAGS = flags.FLAGS
```
[the remark this answers is not in the document]
Comment (jonsimantov) [reply scripts/gha/create_pull_request.py:37]: Mostly it's historical, I think - scripts originally from our internal repo generally use absl flags, scripts that were written as open-source first use either python flags, or, if they are based on an existing script that uses absl flags, absl flags.

Agree that we should at some point standardize - I'd actually love not to use absl-py flags at all at some point in the future, as that would be one less dependency.

**Comment by jonsimantov** [comment]
New sample PR: https://github.com/firebase/firebase-cpp-sdk/pull/646

**Comment by jonsimantov** [comment]
> Double check admob's dependency version.

It's just using the AdMob version set by default in update_android_ios_dependencies.py - when this needs to change, that script should be modified.

File: .github/workflows/update-dependencies.yml (hunk 2 above)
[the remark this answers is not in the document]
Comment (jonsimantov) [reply .github/workflows/update-dependencies.yml]: Actually I think "mobile" is nicer, let's see how that looks.

**Review by DellaBitta — APPROVED** [review]

**Pull request merged**